### ISSUE TITLE
[Bug][Core] GitHub Plugin PAT Token TEST Degraded

### ISSUE THREAD

**Issue by e2corporation** [apache/incubator-devlake#3031]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

GitHub connections are unable to complete a Test operation, API response returns Internal Server 500 instead of expected.

- The GitHub Provider is unable to complete a successful token test even though a valid token is being used. 
- The "invalid" token message is not returned by API personal access tokens cannot be properly validated

<img width="982" alt="Screen Shot 2022-09-08 at 10 20 18 PM" src="https://user-images.githubusercontent.com/1742233/189258545-1f60e0e3-7fb4-40a9-8fab-53b49ec31ca2.png">

![Screen Shot 2022-09-08 at 10 16 45 PM](https://user-images.githubusercontent.com/1742233/189258643-f1ca157b-aa09-4b67-8ce7-ba73e9f1c7a6.png)

<img width="836" alt="Screen Shot 2022-09-08 at 10 22 01 PM" src="https://user-images.githubusercontent.com/1742233/189258879-599b4f9f-5f58-4d58-b612-386346b3f2ef.png">

```
{
  "success": false,
  "message": "error executing the requested resource for plugin github"
}
```

### What you expected to happen

- If a Token Test is **successful** the API responds with expected _true_ response `{ success: true ....}`
- If a Token Test is **unsuccessful** the API responds with _false_ response & applicable message {success: false, message: .... }

### How to reproduce

Test a GitHub Data Connection

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by e2corporation** [comment]
Hey what's up @keon94, this is the GitHub TEST Endpoint I issue we discussed earlier. Thanks.

**Comment by keon94** [comment]
Okay, I'll the expand scope of #2940 and push a fix there since it's already in progress. @e2corporation Does config-ui consume any other endpoints that would return a certain error JSON struct part of the response?

**Comment by keon94** [comment]
And as an idea, probably for another ticket, it might be a good idea to standardize the Backend->Frontend error-message format across all APIs to use some sort of JSON on the response. For instance something like:
```json
{
 "message": "some string",
 "causes": ["cause1", "cause2", "cause3"]
}
```

**Comment by klesh** [comment]
@keon94 It was unified as the following
```json
{
    "success": true|false,
    "message": "errors"
}
```

I think we should keep these 2 fields to maintain FE compatibility as well as other users that call our APIs directly, and add extra fields as you see fit.

**Comment by keon94** [comment]
@e2corporation, what do you think about this output?
![image](https://user-images.githubusercontent.com/25063936/189456904-f59989cd-ef04-4228-b5b3-297cdefef974.png)

**Comment by e2corporation** [comment]
@keon94 It will require updating a couple things on UI but I think that spec could work.

**Comment by keon94** [comment]
@e2corporation #2984 has been merged. Now the backend returns a JSON like the one I mentioned [above](https://github.com/apache/incubator-devlake/issues/3031#issuecomment-1242545195), so if there's any UI side changes that are needed they can be started now.

**Comment by e2corporation** [comment]
> @e2corporation #2984 has been merged. Now the backend returns a JSON like the one I mentioned [above](https://github.com/apache/incubator-devlake/issues/3031#issuecomment-1242545195), so if there's any UI side changes that are needed they can be started now.

ok cool, after I get other core tasks finished I'll have to make some patches on UI to digest the new `causes` key where applicable.

**Comment by klesh** [comment]
Th problem seems to be fixed in main branch bf4de4df3bda079b4cca2ae9e97e8cc9e3d46c54
Close it for now
